core/metrics/application.py

```diff
@@ -17,39 +17,39 @@
 METRICS_CACHE_DURATION = 4*24*60*60  # 4 days (persist over the weekend)
 
 
-def get_application_metrics(application, now_time=None, read_only=False):
-    """
-    Skip image metrics on end-dated applications
-    Otherwise look through the cache to find application metrics
-    """
-    metrics = {}
-    if application.end_date:
-        return metrics
-    metrics = _get_application_metrics(application, interval=rrule.DAILY, now_time=now_time, read_only=read_only)
-    metrics = _get_application_metrics(application, interval=rrule.WEEKLY, now_time=now_time, read_only=read_only)
-    metrics = _get_application_metrics(application, interval=rrule.MONTHLY, now_time=now_time, read_only=read_only)
-    return metrics
-
-
-def _get_application_metrics(application, interval=rrule.MONTHLY, day_limit=120,
-                             now_time=None, force=False, read_only=False):
-    metrics = collections.OrderedDict()
-    if not interval:
-        interval = rrule.MONTHLY
-    redis_cache = redis.StrictRedis()
-    key = "metrics-application-%s-interval-%s-limited-to-%s" % (application.id, rrule.FREQNAMES[interval], day_limit)
-    try:
-        if redis_cache.exists(key) and not force:
-            pickled_object = redis_cache.get(key)
-            metrics = pickle.loads(pickled_object)
-        elif not read_only:
-            metrics = calculate_application_metrics(application, interval, day_limit, now_time=now_time)
-            pickled_object = pickle.dumps(metrics)
-            redis_cache.set(key, pickled_object)
-            redis_cache.expire(key, METRICS_CACHE_DURATION)
-    except:
-        logger.exception("Unexpected errror in application metrics")
-    return metrics
+# def get_application_metrics(application, now_time=None, read_only=False):
```

Comment:
[Delete Your Dead Code](https://late.am/post/2016/04/28/delete-your-dead-code.html)

Reply:
I agree this code should be removed, in a separate PR.. It was commented out in an effort to "pass the tests".